internal/injector/aspect/advice/call.go

```diff
@@ -40,13 +45,65 @@ func (a *appendArgs) Apply(ctx context.Context, chain *node.Chain, csor *dstutil
 		}
 	}
 
-	call.Args = append(call.Args, newArgs...)
+	if !call.Ellipsis {
+		call.Args = append(call.Args, newArgs...)
+		return true, nil
+	}
+
+	// The function call has an ellipsis, so we need to append our new arguments to the last argument,
+	// which is a slice. To do so, we need to provision a new slice of the right type and size, append
+	// all the relevant data in there, and then replace the last argument with the new slice.
+	lastIdx := len(call.Args) - 1
+	call.Args[lastIdx] = &dst.CallExpr{
+		Fun: &dst.FuncLit{
+			Type: &dst.FuncType{
+				Params: &dst.FieldList{
+					List: []*dst.Field{{
+						Names: []*dst.Ident{dst.NewIdent("opts")},
+						Type:  &dst.Ellipsis{Elt: a.typeName.AsNode()},
+					}},
+				},
+				Results: &dst.FieldList{
+					List: []*dst.Field{{
+						Type: &dst.ArrayType{Elt: a.typeName.AsNode()},
+					}},
+				},
+			},
+			Body: &dst.BlockStmt{
+				List: []dst.Stmt{
+					&dst.ReturnStmt{Results: []dst.Expr{
+						&dst.CallExpr{
+							Fun: dst.NewIdent("append"),
+							Args: append(
+								append(
+									make([]dst.Expr, 0, len(newArgs)+1),
+									dst.NewIdent("opts"),
```

Comment:
how do you know the identifier used in the original func call is opts? 🤔 Shouldn't you reuse the argument's AST instead in order to retrieve the identifier?

Reply:
This is an argument provided to the IIFE, and I chose the name for it...